pkg/handler/snapshots.go

```diff
@@ -206,3 +224,211 @@ func (sh *SnapshotHandler) getRepoConfigurationFile(c echo.Context) error {
+		snap, err := sh.DaoRegistry.Snapshot.Fetch(c.Request().Context(), snapshotUUIDs[i])
+		responses[i] = snap
+		if err != nil {
+			hasErr = true
+			errs[i] = err
+			continue
+		}
+		err = sh.isSnapInRepo(repoSnaps, snap.UUID)
+		if err != nil {
+			hasErr = true
+			errs[i] = err
+			continue
+		}
+	}
+	if hasErr {
+		return ce.NewErrorResponseFromError("Error deleting snapshots", errs...)
+	}
+
+	errs = sh.DaoRegistry.Snapshot.BulkDelete(c.Request().Context(), snapshotUUIDs)
+	if len(errs) > 0 {
+		return ce.NewErrorResponseFromError("Error deleting snapshots", errs...)
+	}
+
+	enqueueErr := sh.enqueueDeleteSnapshotsTask(c, orgID, repoUUID, snapshotUUIDs...)
+	if enqueueErr != nil {
+		hasErr = false
+		errs = make([]error, len(snapshotUUIDs))
+		for i := range snapshotUUIDs {
+			err = sh.DaoRegistry.Snapshot.ClearDeletedAt(c.Request().Context(), snapshotUUIDs[i])
+			if err != nil {
+				hasErr = true
+				errs[i] = err
+			}
+		}
+		if hasErr {
+			return ce.NewErrorResponseFromError("Error clearing snapshot deleted_at fields", errs...)
+		}
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(enqueueErr), "Error enqueueing task", enqueueErr.Error())
+	}
+
+	return c.NoContent(http.StatusNoContent)
+}
+
+func (sh *SnapshotHandler) enqueueDeleteSnapshotsTask(c echo.Context, orgID, repoUUID string, snapshotUUIDs ...string) error {
+	accountID, _ := getAccountIdOrgId(c)
+	payload := payloads.DeleteSnapshotsPayload{RepoUUID: repoUUID, SnapshotsUUIDs: snapshotUUIDs}
+	task := queue.Task{
+		Typename:   config.DeleteSnapshotsTask,
+		Payload:    payload,
+		OrgId:      orgID,
+		AccountId:  accountID,
+		ObjectUUID: utils.Ptr(repoUUID),
+		ObjectType: utils.Ptr(config.ObjectTypeRepository),
+		RequestID:  c.Response().Header().Get(config.HeaderRequestId),
+	}
+
+	taskID, err := sh.TaskClient.Enqueue(task)
+	if err != nil {
+		logger := tasks.LogForTask(taskID.String(), task.Typename, task.RequestID)
+		logger.Error().Msg("error enqueuing task")
+		return err
+	}
+
+	return nil
+}
+
+func (sh *SnapshotHandler) isDeleteInProgress(c echo.Context, orgID, repoUUID string) error {
+	inProgressTasks, err := sh.DaoRegistry.TaskInfo.
+		FetchActiveTasks(c.Request().Context(), orgID, repoUUID, config.DeleteRepositorySnapshotsTask, config.DeleteSnapshotsTask)
+
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching delete repository snapshots task", err.Error())
+	}
+	if len(inProgressTasks) >= 1 {
+		return ce.NewErrorResponse(http.StatusBadRequest, "Error deleting snapshot", "Delete is already in progress")
+	}
+
+	return nil
+}
+
+func (sh *SnapshotHandler) isDeleteAllowed(c echo.Context, orgID, repoUUID string, snapshotUUIDs ...string) ([]models.Snapshot, error) {
+	_, err := sh.DaoRegistry.RepositoryConfig.Fetch(c.Request().Context(), orgID, repoUUID)
+	if err != nil {
+		return nil, ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository config", err.Error())
+	}
```

Comment:
One thing we also need to for this is disallow users from deleting snapshots for red hat repos. Right now, if I use the repo UUID and snapshot UUID of a red hat repo and snapshot, I can delete a snapshot for a red hat repo.

Reply:
I think ideally it'd return a 404 and just say "repo not found"

Reply:
good point, missed that check in the handler 👍🏼